### PR TITLE
Get HummingbirdFluent working with HB 2.0

### PR DESCRIPTION
**Pull request by adam-fowler**
Remove HBApplication extensions
HBFlluent is now a Service.
Make HBFluent Sendable by 
- Making Migrations only accessible on MainActor.
- Wrapping Databases whose access is already protected by a lock in an UnsafeTransfer.

Update HBFluentPersistDriver

### CODE REVIEW

File: Sources/HummingbirdFluent/Fluent.swift [hunk 1]
```diff
@@ -14,85 +14,57 @@
...
+public struct HBFluent: Sendable, Service {
+    /// Event loop group
     public let eventLoopGroup: EventLoopGroup
     /// Logger
     public let logger: Logger
-    /// Fluent history setup
-    public let history: History
+    /// List of migrations. Only accessible from the main actor
+    @MainActor
+    public var migrations: Migrations { self._migrations.value }
```
Comment (adam-fowler) [inline Sources/HummingbirdFluent/Fluent.swift:37]: I'm currently using `@MainActor` here. @Joannis what are your thoughts on setting up a Hummingbird global actor in Hummingbird and using that instead.

Comment (Joannis) [reply Sources/HummingbirdFluent/Fluent.swift:37]: Do you foresee other places where you'd want/need this? What's the "domain" of the actor? What type of information do you scope it to?

File: Sources/HummingbirdFluent/Persist+fluent.swift [hunk 2]
```diff
@@ -12,151 +12,110 @@
...
-                    }
-                    throw error
-                }
-                .map { _ in }
+            try await model.save(on: db)
+        } catch let error as DatabaseError where error.isConstraintFailure {
+            throw HBPersistError.duplicate
         } catch {
-            return request.eventLoop.makeFailedFuture(error)
+            print("\(error)")
```
Comment (Joannis) [inline Sources/HummingbirdFluent/Persist+fluent.swift:52]: Print statement

File: Sources/HummingbirdFluent/Persist+fluent.swift [hunk 3]
```diff
@@ -12,151 +12,110 @@
...
+                model.data = data
+                model.expires = date
+                try await model.update(on: db)
+            } else {
+                let model = PersistModel(id: key, data: data, expires: date)
+                try await model.save(on: db)
+            }
         } catch {
-            return request.eventLoop.makeFailedFuture(error)
+            print("\(error)")
```
Comment (Joannis) [inline Sources/HummingbirdFluent/Persist+fluent.swift:78]: Print statement

**Review by Joannis — COMMENTED** [review]
I haven't worked on or with Fluent for ages, so I'm not confident I can give a great review when it comes to API design

File: Sources/HummingbirdFluent/Fluent.swift (hunk 1 above)
Comment (adam-fowler) [reply Sources/HummingbirdFluent/Fluent.swift:37]: At this point I time probably not so we can keep as `@MainActor`. The migration stuff should only be setup during initialisation.

**Comment by adam-fowler** [comment]
> I haven't worked on or with Fluent for ages, so I'm not confident I can give a great review when it comes to API design

It's not like I've used it a great deal. I'm not really a fan. I basically provided this as I know a lot of people rely on it.

**Pull request merged**